webapp/channels/src/selectors/admin_console.tsx

```diff
@@ -32,14 +32,15 @@ export const getConsoleAccess = createSelector(
     getMySystemPermissions,
     (adminDefinition, mySystemPermissions) => {
         const consoleAccess = {read: {}, write: {}};
-        const addEntriesForKey = (entryKey) => {
+        const addEntriesForKey = (entryKey: string) => {
             const permissions = ResourceToSysConsolePermissionsTable[entryKey].filter((x) => mySystemPermissions.has(x));
-            consoleAccess.read[entryKey] = permissions.length !== 0;
-            consoleAccess.write[entryKey] = permissions.some((permission) => permission.startsWith('sysconsole_write_'));
+            Object.assign(consoleAccess.read, {[entryKey]: permissions.length !== 0});
+            Object.assign(consoleAccess.write, {[entryKey]: permissions.some((permission) => permission.startsWith('sysconsole_write_'))});
         };
-        const mapAccessValuesForKey = ([key]) => {
-            if (typeof RESOURCE_KEYS[key.toUpperCase()] === 'object') {
-                Object.values(RESOURCE_KEYS[key.toUpperCase()]).forEach((entry) => {
+        const mapAccessValuesForKey = ([key]: [string, any]) => {
+            const upperKey = key.toUpperCase() as keyof typeof RESOURCE_KEYS;
```

Comment:
Nothing to change here, but I do really wonder why these permissions aren't part of the schema definition themselves. This being miserable to type feels like TS is telling us we're doing something wrong 😛